tahoe_sites/api.py

```diff
@@ -236,15 +237,15 @@ def get_uuid_by_site(site):
     return TahoeSite.objects.get(site=site).site_uuid
 
 
-def get_current_site(request):
+def get_current_site(request=None):
```

Comment:
I don't think this is a good function design.

I think we probably should be consistent with Open edX on this specific matter:


```
def get_current_request():
    """
    Return current request instance.

    Returns:
         (HttpRequest): returns current request
    """
    return crum.get_current_request()


def get_current_site():
    """
    Return current site.

    Returns:
         (django.contrib.sites.models.Site): returns current site
    """
    request = get_current_request()
    if not request:
        return None
    return getattr(request, 'site', None)
```


If you really want to, you can make a third mini function `get_site_by_request(request)` which does `getattr(request, 'site', None)`

Reply:
Done